pkg/ee/cluster-backup/controller.go

```diff
@@ -234,3 +245,69 @@ func (r *reconciler) ensureClusterBackupUserClusterResources(ctx context.Context
 
 	return nil
 }
+
+func (r *reconciler) undeployClusterBackupUserClusterComponents(ctx context.Context, cluster *kubermaticv1.Cluster) error {
+	userClusterClient, err := r.userClusterConnectionProvider.GetClient(ctx, cluster)
+	if err != nil {
+		return fmt.Errorf("failed to get user cluster client: %w", err)
+	}
+	if err := r.undeployClusterBackupUserClusterResources(ctx, userClusterClient); err != nil {
+		return fmt.Errorf("failed to undeploy cluster backup user-cluster resources: %w", err)
+	}
+	return r.undeployClusterBackupUserClusterCRDs(ctx, userClusterClient)
+}
+
+func (r *reconciler) undeployClusterBackupUserClusterResources(ctx context.Context, userClusterClient ctrlruntimeclient.Client) error {
+	userClusterResources := []ctrlruntimeclient.Object{
+		&appsv1.DaemonSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      userclusterresources.DaemonSetName,
+				Namespace: resources.ClusterBackupNamespaceName,
+			},
+		},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      userclusterresources.DeploymentName,
+				Namespace: resources.ClusterBackupNamespaceName,
+			},
+		},
+		// The rest of the resources are non-workload resources. Deleting the namespace should take care of them.
+		&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: resources.ClusterBackupNamespaceName,
```

Comment:
Please delete all the resources explicitly instead of relying on the namespace cleanup.